### PR TITLE
Update some commentary.

### PR DESCRIPTION
**Pull request by bangerth**
Part of #14403. Basically a follow-up to #14421.

/rebuild

### CODE REVIEW

**Review by tamiko — APPROVED** [review]

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -4989,17 +4989,16 @@ namespace DataOutBase
...
-    // write them one variable at a time and don't want to use more than one
-    // loop
-    //
-    // this copying of data vectors can be done while we already output the
-    // vertices, so do this on a separate task and when wanting to write out the
+    // An easier way is to create a global table that for each variable
+    // lists all values. This copying of data vectors can be done in the
+    // background while we're already working on vertices and cells,
+    // so do this on a separate task and when wanting to write out the
     // data, we wait for that task to finish
```
Comment (tamiko) [inline source/base/data_out_base.cc:5002]:
```suggestion
    // data, we wait for that task to finish.
```

File: source/base/data_out_base.cc [hunk 2]
```diff
@@ -5165,19 +5164,17 @@ namespace DataOutBase
...
+    // variable we're dealing with right now, and then start the process over
+    // for the next variable with another loop over all patches.
     //
-    // this copying of data vectors can be done while we already output the
-    // vertices, so do this on a separate task and when wanting to write out the
+    // An easier way is to create a global table that for each variable
+    // lists all values. This copying of data vectors can be done in the
+    // background while we're already working on vertices and cells,
+    // so do this on a separate task and when wanting to write out the
     // data, we wait for that task to finish
```
Comment (tamiko) [inline source/base/data_out_base.cc:5177]:
```suggestion
    // data, we wait for that task to finish.
```

File: source/base/data_out_base.cc [hunk 3]
```diff
@@ -5401,17 +5398,17 @@ namespace DataOutBase
...
```
Comment (tamiko) [inline source/base/data_out_base.cc:5412]:
```suggestion
    // data, we wait for that task to finish.
```

File: source/base/data_out_base.cc [hunk 4]
```diff
@@ -5702,17 +5699,16 @@ namespace DataOutBase
...
```
Comment (tamiko) [inline source/base/data_out_base.cc:5712]:
```suggestion
    // data, we wait for that task to finish.
```

File: source/base/data_out_base.cc [hunk 5]
```diff
@@ -6079,17 +6075,16 @@ namespace DataOutBase
...
```
Comment (tamiko) [inline source/base/data_out_base.cc:6088]:
```suggestion
    // data, we wait for that task to finish.
```

File: source/base/data_out_base.cc [hunk 6]
```diff
@@ -9151,6 +9146,17 @@ DataOutBase::write_filtered_data(
...
+  // to one variable at a time. We could in principle do this by looping
+  // over all patches and extracting the values corresponding to the one
+  // variable we're dealing with right now, and then start the process over
+  // for the next variable with another loop over all patches.
+  //
+  // An easier way is to create a global table that for each variable
+  // lists all values. This copying of data vectors can be done in the
+  // background while we're already working on vertices and cells,
+  // so do this on a separate task and when wanting to write out the
+  // data, we wait for that task to finish
```
Comment (tamiko) [inline source/base/data_out_base.cc:9159]:
```suggestion
  // data, we wait for that task to finish.
```

**Comment by bangerth** [comment]
So updated.

**Pull request merged**